src/Deposit.sol

```diff
@@ -0,0 +1,94 @@
+pragma solidity ^0.8.13;
+
+import {Ownable} from "solady/src/auth/Ownable.sol";
+import {SafeTransferLib} from "solady/src/utils/SafeTransferLib.sol";
+import {IEthStrategy} from "./DutchAuction.sol";
+import {SignatureCheckerLib} from "solady/src/utils/SignatureCheckerLib.sol";
+import {TReentrancyGuard} from "../lib/TReentrancyGuard/src/TReentrancyGuard.sol";
+
+contract Deposit is Ownable, TReentrancyGuard {
+
+  uint256 public immutable CONVERSION_RATE;
+  uint256 constant MIN_DEPOSIT = 1e18;
+  uint256 constant MAX_DEPOSIT = 100e18;
+
+  error DepositAmountTooLow();
+  error DepositAmountTooHigh();
+  error DepositFailed();
+  error AlreadyRedeemed();
+  error DepositCapExceeded();
+  error InvalidConversionPremium();
+  error InvalidSignature();
+  error InvalidCall();
+
+  address public immutable ethStrategy;
+  address public signer;
+
+  mapping(address => bool) public hasRedeemed;
+
+  uint256 public immutable conversionPremium;
```

Comment:
if you are considering changing the fee in the future you might want to reconsider making it immutable

Reply:
This was purposeful so there is optionality in the future but will deploy a new instance if we want to change it